apps/gitness/src/pages/project-settings/project-settings-new-member-page.tsx

```diff
@@ -0,0 +1,214 @@
+  DropdownMenuContent,
+  DropdownMenuLabel,
+  DropdownMenuSeparator,
+  Spacer,
+  Text,
+  Icon,
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue
+} from '@harnessio/canary'
+import { SandboxLayout, FormFieldSet } from '@harnessio/playground'
+import { useNavigate } from 'react-router-dom'
+import { useGetSpaceURLParam } from '../../framework/hooks/useGetSpaceParam'
+import { useMembershipAddMutation, EnumMembershipRole, useListPrincipalsQuery } from '@harnessio/code-service-client'
+
+// Define form schema for new member
+const newMemberSchema = z.object({
+  memberName: z.string().min(1, { message: 'Please provide a member name' }),
+  role: z.string().min(1, { message: 'Please select a role for the new member' })
+})
+
+// TypeScript types for form fields
+type NewMemberFields = z.infer<typeof newMemberSchema>
+
+// Role options as string values, compatible with the API
+const roleOptions = [
+  { label: 'Owner', value: 'space_owner' },
+  { label: 'Reader', value: 'reader' },
+  { label: 'Executor', value: 'executor' },
+  { label: 'Contributor', value: 'contributor' }
+]
+
+export const CreateNewMemberPage = () => {
+  const space_ref = useGetSpaceURLParam()
+  const navigate = useNavigate()
+  const [selectedMember, setSelectedMember] = useState<string>('') // State to hold selected member
+
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    reset: resetNewMemberForm,
+    formState: { errors, isValid },
+    watch
+  } = useForm<NewMemberFields>({
+    resolver: zodResolver(newMemberSchema),
+    mode: 'onChange',
+    defaultValues: {
+      memberName: '',
+      role: ''
+    }
+  })
+
+  // Fetch available members
+  // This type in custom hook show: array type but not work here. we can only use string to get api data
+  const { data: { body: usersData } = {} } = useListPrincipalsQuery({
+    queryParams: {
+      page: 1,
+      limit: 20,
+      type: 'user'
+    }
+  })
+
+  // Mutation hook for adding a member
+  const {
+    mutate: addMember,
+    isSuccess: submitted,
+    isLoading: isSubmitting
+  } = useMembershipAddMutation(
+    { space_ref },
+    {
+      onSuccess: () => {
+        resetNewMemberForm()
+        navigate(`/spaces/${space_ref}/settings/members`, { replace: true })
+      },
+      onError: error => {
+        alert('Error adding member: ' + error.message)
+      }
+    }
+  )
+
+  // Form submit handler
+  const onSubmit: SubmitHandler<NewMemberFields> = data => {
+    addMember({ body: { user_uid: data.memberName, role: data.role as EnumMembershipRole } })
+  }
+
+  // Handle member selection
+  const handleMemberSelect = (uid: string, display_name: string) => {
+    setSelectedMember(display_name)
+    setValue('memberName', uid, { shouldValidate: true })
+  }
+
+  const handleCancel = () => {
+    resetNewMemberForm()
+    navigate(`/spaces/${space_ref}/settings/members`, { replace: true })
+  }
+
+  const newMemberRoleValue = watch('role')
+
+  return (
```

Comment:
this component should be in `playground`. Only the API calls should be in `gitness`.

Reply:
since this page will be revised again so I decide not affect this page in the playground, hope team members can follow up if the flow of combo box component has been decided